### PR TITLE
feat: touch up callout component

### PR DESCRIPTION
**Pull request by Creeland**
Made a few tweaks to the Callout component:

Added text prop for easier text rendering, falls back to children if not provided. I noticed that the callout had some whitespace due to "children" not just being the text, but the html that was created from the text when the markdown was compiled. 

I also cleaned up the component and added some additional text styling. 

## Old

image
image

## New

image
image

### CODE REVIEW

File: src/components/mdx/callout.js [hunk 1]
```diff
@@ -1,13 +1,9 @@
-import React from 'react'
-
-const container = ''
-
-export default function Callout({children, className}) {
+export default function Callout({children, text, className}) {
   return (
     <div
-      className={`mx-auto my-3 w-full dark:bg-gray-800 bg-blue-100 border-l-4 overflow-hidden border-blue-900 dark:text-blue-200 text-blue-600 sm:px-8 px-4 py-3 rounded-sm`}
+      className={`leading-relaxed mx-auto my-3 w-full dark:bg-gray-800 bg-blue-100 border-l-4 border-blue-900  sm:px-8 px-4 py-6 rounded-sm font-small sm:text-lg text-base tracking-normal shadow-md ${className}`}
```
Comment (vojtaholik) [inline src/components/mdx/callout.js:4]: `font-small` is not a valid tailwind class

**Comment by Creeland** [comment]
Updated the component to be more consistent with our horizontal resource card spacing. I also dropped the class that didn't exist that I had mistakenly used @vojtaholik 

image
  
image

File: src/components/mdx/callout.js [hunk 2]
```diff
@@ -1,13 +1,9 @@
-import React from 'react'
-
-const container = ''
-
-export default function Callout({children, className}) {
+export default function Callout({children, text, className}) {
   return (
     <div
-      className={`mx-auto my-3 w-full dark:bg-gray-800 bg-blue-100 border-l-4 overflow-hidden border-blue-900 dark:text-blue-200 text-blue-600 sm:px-8 px-4 py-3 rounded-sm`}
+      className={`p-5 leading-relaxed mx-auto my-3 w-full dark:bg-gray-800 bg-blue-100 border-l-4 border-blue-900 sm:space-x-5 space-x-0 sm:space-y-0 space-y-5 items-center text-left shadow-sm rounded-lg overflow-hidden lg:text-lg sm:text-md ${className}`}
```
Comment (vojtaholik) [inline src/components/mdx/callout.js:4]: using `className` like this without making it an empty string as a fallback will make it return `undefined` by default.

File: src/components/mdx/callout.js [hunk 3]
```diff
@@ -1,13 +1,9 @@
-import React from 'react'
-
-const container = ''
-
-export default function Callout({children, className}) {
+export default function Callout({children, text, className}) {
```
Comment (vojtaholik) [inline src/components/mdx/callout.js:1]: why not convert it to TS while you're there? :)

**Comment by vojtaholik** [comment]
I used actual links, fixed horizontal scrolling caused by code blocks by reducing their negative horizontal margin, and made the callouts stretch to full width on mobile.

screenshots

File: src/components/mdx/callout.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  return (
+    <div className="sm:mx-auto -mx-2">
+      <div
+        className={twMerge(
+          'sm:py-5 sm:px-5 py-3 px-5 leading-relaxed my-3 w-full dark:bg-gray-800 bg-blue-200/20 border-l-4 dark:border-blue-600 border-blue-500 items-center text-left shadow-sm sm:rounded-md overflow-hidden lg:text-lg sm:text-md',
+          className,
+        )}
+        aria-label="callout"
+      >
+        {text ? <p>{text}</p> : children}
```
Comment (vojtaholik) [inline src/components/mdx/callout.tsx:23]: we want paragraphs for seo and a11y

**Review by vojtaholik — APPROVED** [review]

**Pull request merged**